iroh-metrics/src/service.rs

```diff
@@ -1,43 +1,45 @@
-use std::{future::Future, io, net::SocketAddr, pin::Pin};
+use std::net::SocketAddr;
 
-use hyper::{
-    service::{make_service_fn, service_fn},
-    Body, Error, Request, Response, Server,
-};
-
-use tracing::info;
+use anyhow::{anyhow, Result};
+use hyper::service::service_fn;
+use hyper::{Request, Response};
+use tokio::net::TcpListener;
+use tracing::{error, info};
 
 use crate::core::Core;
 
+type BytesBody = http_body_util::Full<hyper::body::Bytes>;
+
 /// Start a HTTP server to report metrics.
-pub async fn run(metrics_addr: SocketAddr) -> Result<(), Error> {
+pub async fn run(metrics_addr: SocketAddr) -> Result<()> {
     info!("Starting metrics server on {metrics_addr}");
-    Server::bind(&metrics_addr)
-        .serve(make_service_fn(move |_conn| async move {
-            let handler = make_handler();
-            Ok::<_, io::Error>(service_fn(handler))
-        }))
-        .await
+    let listener = TcpListener::bind(metrics_addr).await?;
+    loop {
+        let (stream, _addr) = listener.accept().await?;
+        let io = hyper_util::rt::TokioIo::new(stream);
+        tokio::spawn(async move {
```

Comment:
why do we need to spawn now?

Reply:
Because we need to write the accepting loop by hand, previously `Server.serve()` did the spawning but they took away the whole `Server` thing and are only given lower-level bits now.

Reply:
I see